src/ray/raylet/scheduling_policy.cc

```diff
@@ -60,12 +56,18 @@ std::unordered_map<TaskID, ClientID> SchedulingPolicy::Schedule(
       // TODO(atumanov): change uniform random to discrete, weighted by resource capacity.
       std::uniform_int_distribution<int> distribution(0, client_keys.size() - 1);
       int client_key_index = distribution(gen_);
-      decision[task_id] = client_keys[client_key_index];
+      const ClientID &dst_client_id = client_keys[client_key_index];
+      // Decrement destination client's available resources at placement decision time.
+      cluster_resources[dst_client_id].Acquire(
+          t.GetTaskSpecification().GetRequiredResources());
+      // Fractional and enumerated resources will be acquired on task dispatch.
```

Comment:
Why separate the behavior of `cluster_resources` from `local_available_resources_`?

Reply:
In this case, I'm trying to use `cluster_resources` to keep track of "committed" resources, which keep track of how much capacity was _allocated_ , while local_available_resources_ is updated only when the task is actually dispatched and done, keeping track of _local_ resources precisely.

Reply:
no, that would not be addressed by making a temporary local copy of `cluster_resources`, because I need to keep track of the number of tasks locally placed across invocations of the policy.
Yes, I do want to leave open the possibility of refactoring `cluster_resources` to subsume the thing you added by introducing a polymorphic `Resource` type.

Reply:
I am working on the fix to this, that uses local load information in placement decisions instead of keeping the load information as part of the `cluster_resources` data structure. This is in progress.